packages/manager/apps/pci-public-ip/src/pages/floating-ips/edit/EditInstance.page.tsx

```diff
@@ -195,11 +204,12 @@ export default function EditInstancePage() {
                     ))}
                   </OsdsSelect>
                 </OsdsFormField>
-                {!selectedPrivateNetwork?.gatewayIP && (
+                {!selectedPrivateNetwork?.gatewayIp && (
                   <OsdsMessage
                     type={ODS_MESSAGE_TYPE.warning}
                     color={ODS_THEME_COLOR_INTENT.warning}
                     className="mt-8"
+                    data-testid="editInstancePage_message_noGatewayIPFound"
```

Comment:
you could use getByText instead of using testid

Reply:
I avoid as possible as i can to retrieve element by selector that may change after, that's why i prefer to add data-testid to avoid such as issue, and simplify the retrieve of my element in the same time. 
And data-testid are made for this purpose ;)

Reply:
https://testing-library.com/docs/queries/bytestid/

```
In the spirit of [the guiding principles](https://testing-library.com/docs/guiding-principles), it is recommended to use this only after the other queries don't work for your use case. Using data-testid attributes do not resemble how your software is used and should be avoided if possible. That said, they are way better than querying based on DOM structure or styling css class names. Learn more about data-testids from the blog post ["Making your UI tests resilient to change"](https://kentcdodds.com/blog/making-your-ui-tests-resilient-to-change)
```